code/modules/mob/living/simple_animal/hostile/statue.dm

```diff
@@ -53,8 +53,8 @@
 
 	var/cannot_be_seen = 1
 	var/mob/living/creator = null
-
-
+	///ability to phase in and out when unseen.
+	var/datum/action/innate/creature/teleport/teleport
```

Comment:
Handle this on `Destroy()`

Reply:
Should be weakreffed actually, actions can auto delete if their target is removed